File: virtual-store/backend/src/main/java/com/dev/backend/model/Product.java
package com.dev.backend.model;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;

@EqualsAndHashCode(callSuper = true)
@Data
@Entity
public class Product extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String name;

    @Lob
    private String shortDescription;

    @Lob
    private String detailedDescription;

    @ManyToOne
    private Brand brand;

    @ManyToOne
    private Category category;

    private Double costValue;

    private Double saleValue;

    @OneToOne
    private Image image;
}

File: virtual-store/frontend/src/model/Product.ts
import {Brand} from "./Brand";
import {Category} from "./Category";
import {Image} from "./Image";

export class Product {
    id?: number;
    name?: string;
    shortDescription?: string;
    detailedDescription?: string;
    brand?: Brand;
    category?: Category;
    costValue?: number;
    saleValue?: number;
    image?: Image;
}